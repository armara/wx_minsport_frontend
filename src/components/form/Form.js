import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Form as FinalForm } from 'react-final-form'
import arrayMutators from 'final-form-arrays'

import 'components/form/Form.scss'

const Form = ({
  validate,
  onSubmit,
  decorators,
  initialValues,
  subscription,
  className,
  children,
  childrenProps,
  ...rest
}) => (
  <FinalForm
    onSubmit={onSubmit}
    validate={validate}
    decorators={decorators}
    initialValues={initialValues}
    mutators={{
      ...arrayMutators,
    }}
  >
    {({ handleSubmit, form }) => {
      return (
        <form
          onSubmit={handleSubmit}
          className={classnames([className, 'base-form'])}
        >
          {typeof children === 'function'
            ? children({ form, initialValues, ...childrenProps, ...rest })
            : children}
        </form>
      )
    }}
  </FinalForm>
)

Form.defaultProps = {
  className: '',
  initialValues: {},
  childrenProps: {},
  subscription: {},
  decorators: [],
  validate: () => {},
}

Form.propTypes = {
  decorators: PropTypes.arrayOf(PropTypes.func),
  subscription: PropTypes.shape({}),
  className: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  validate: PropTypes.func,
  childrenProps: PropTypes.shape({}),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.func,
  ]).isRequired,
  initialValues: PropTypes.shape({}),
}

export default Form
