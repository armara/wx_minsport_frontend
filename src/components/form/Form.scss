@import "styles/mixins";

.base-form {
  display: flex;
  flex-flow: column;

  h4 {
    @include form-label-text();
  }
}

.form-some-blocks {
  display: flex;

  & > div {
    flex-grow: 1;
  }

  & > div + div {
    margin-left: 10px;
  }
}
