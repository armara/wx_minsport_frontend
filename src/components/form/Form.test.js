import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Form from 'components/form/Form'

configure({ adapter: new Adapter() })

describe('<Form />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(
        <Form onSubmit={() => {}} validate={() => {}}>
          test
        </Form>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
