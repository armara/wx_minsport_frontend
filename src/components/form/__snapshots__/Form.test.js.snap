// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders correctly default props 1`] = `
<form
  className="base-form"
  onSubmit={[Function]}
>
  test
</form>
`;
